lib/member_page.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+require 'scraped'
+require 'pry'
+
+# This class represents the profile page of a given member
+class MemberPage < Scraped::HTML
+  decorator Scraped::Response::Decorator::CleanUrls
+
+  field :id do
+    url.gsub('.htm', '').split('members/').last
+  end
+
+  field :name do
+    name_parts.first.to_s.gsub(Regexp.union(titles << '.'), '').tidy
```

Comment:
it's not really clear what this `titles << '.'` is doing, but the tests pass without it, so if it's important I think you need an extra test case.

This approach also seems flawed, in that it's going to fail if anyone ever has a name that contains one of the prefixes (e.g. "Draper").

More generally, this is already a solved problem in other scrapers — why are you reinventing a new solution to it rather than reusing an existing approach?